genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/ServicesAutoConfiguration.java

```diff
@@ -107,4 +110,27 @@ public KillService killService(final ApplicationEventPublisher applicationEventP
     public LaunchJobService launchJobService() {
         return new LaunchJobServiceImpl();
     }
+
+    /**
+     * Provide a lazy {@link ArchivalService} bean in case no AmazonS3 client can be created.
+     * @return A {@link ArchivalService} instance
+     */
+    @Bean
+    @Lazy
+    @ConditionalOnClass(value = AmazonS3.class)
+    public ArchivalService archivalService() {
+        return new DummyArchivalServiceImpl();
+    }
+
+    /**
+     * Provide a lazy {@link ArchivalService} bean if one hasn't already been defined.
+     * @param amazonS3 Amazon S3 client instance
+     * @return A {@link ArchivalService} instance
+     */
+    @Bean
+    @Lazy
+    @ConditionalOnMissingBean(ArchivalService.class)
```

Comment:
`@ConditionalOnBean(AmazonS3.class)`

Reply:
Hmm.. I see the problem. I wrongly assumed that the beans with `@ConditionalOnMissingBean` are evaluated in the end. Will fix.

Reply:
I take that back. What i had was totally wrong and not just a bad assumption. Fixed it

Reply:
Done